### PR TITLE
Fix comment duplication in config files on 1.18

### PR DESCRIPTION
**Pull request by nicegamer7**
This PR fixes comment duplication in config files on 1.18 (#480) by only adding comments when the file is first created.

1.18 adds support for comments, and so it's no longer necessary to add comments every time the server starts. This was the reason for the "duplication" (not actually duplication, it was growing by one line at a time).

Initially I made our wrapper class inherit from YamlConfiguration (which is why I wanted #361 merged, it made that much easier imo), and then I initialized the config objects to either YamlConfiguration or our wrapper class, depending on the server's version. However, I learned afterwards that we still need to add comments when the file is first created, and that is where this solution came from.

### CODE REVIEW

File: src/main/java/com/onarandombox/multiverseinventories/YamlWorldGroupManager.java [hunk 1]
```diff
@@ -30,19 +30,21 @@
...
-        if (!groupConfigFile.exists()) {
+        if (!groupsConfigFile.exists()) {
             Logging.fine("Created groups file.");
-            groupConfigFile.createNewFile();
+            groupsConfigFile.createNewFile();
             migrateGroups = true;
         }
         // Load the configuration file into memory
-        groupsConfig = new CommentedYamlConfiguration(groupConfigFile, true);
+        boolean supportsCommentsNatively = Integer.parseInt(this.plugin.getServer().getBukkitVersion().split("\\.")[1]) > 17;
```
Comment (benwoo1110) [inline src/main/java/com/onarandombox/multiverseinventories/YamlWorldGroupManager.java:46]: Generally, i think @dumptruckman dont really prefer version string check to apply different behaviour, but I think in this case likely the easiest soln.

**Review by benwoo1110 — COMMENTED** [review]
Tested on my end, comment duplication looks to be fixed. Test build for others to try: 
[Multiverse-Inventories-4.2.3-SNAPSHOT.zip](https://github.com/Multiverse/Multiverse-Inventories/files/7847828/Multiverse-Inventories-4.2.3-SNAPSHOT.zip)

File: src/main/java/com/onarandombox/multiverseinventories/YamlWorldGroupManager.java (hunk 1 above)
Comment (mikeespain) [reply src/main/java/com/onarandombox/multiverseinventories/YamlWorldGroupManager.java:46]: I'm still getting the same message in my console:


[14:23:53] [Server thread/ERROR]: Error occurred while enabling Multiverse-Inventories v4.2.3-SNAPSHOT-bUNKNOWN (Is it up to date?)
java.lang.StackOverflowError: null
	at org.yaml.snakeyaml.parser.ParserImpl$ParseBlockMappingValueComment.produce(ParserImpl.java:684) ~[snakeyaml-1.30.jar:?]



Followed by another 1,023 lines of the same thing. My server then disables Multiverse-Inventories.

Is the fix meant to be just a copy and paste the jar situation?


Thanks for your hard work and all the devs! Appreciate the support

Comment (nicegamer7) [reply src/main/java/com/onarandombox/multiverseinventories/YamlWorldGroupManager.java:46]: This PR fixes the issue of comment duplication. But it doesn't remove the comments that were already added.

You'll need to open the two config files and remove the duplicate comments to get it running again.

Comment (nicegamer7) [reply src/main/java/com/onarandombox/multiverseinventories/YamlWorldGroupManager.java:46]: > Generally, i think @dumptruckman dont really prefer version string check to apply different behaviour, but I think in this case likely the easiest soln.

I'm not particularly fond of it either, but I didn't find any other way to do it. Open to suggestions.

Comment (mikeespain) [reply src/main/java/com/onarandombox/multiverseinventories/YamlWorldGroupManager.java:46]: > This PR fixes the issue of comment duplication. But it doesn't remove the comments that were already added.
> 
> You'll need to open the two config files and remove the duplicate comments to get it running again.

Thanks for that! I tried just deleting the entry, and it came up with the same error, so I just deleted the two config files and started again and it's working fine. Cheers!

Comment (benwoo1110) [reply src/main/java/com/onarandombox/multiverseinventories/YamlWorldGroupManager.java:46]: yea, that will be a better option. PaperLib  should be a small dependency

**Comment by nicegamer7** [comment]
Updated to use PaperLib, should be good to go now.

I think we should integrate PaperLib into Core for V5. We can use it for things like async teleports.

**Review by benwoo1110 — APPROVED** [review]
👍

**Pull request merged**